csharp/src/Google.Protobuf.Test/GeneratedMessageTest.cs

```diff
@@ -111,6 +112,50 @@ public void DefaultValues()
             Assert.AreEqual("", message.OneofString);
             Assert.AreEqual(ByteString.Empty, message.OneofBytes);
             Assert.IsNull(message.OneofNestedMessage);
+
+            // proto2 default values

```

Comment:
For the tests that proto2 and proto3 have same behavior, we should add test for both of them. For example  the above "Assert.AreEqual(false, message.SingleBool); ..." should be tested for both proto2 and proto3. Create the message seperately and then call common  assert code.  

For better read ability, we can put the the common tests together (for example in the front), put proto2 only tests together(middle) and proto3 only tests together (in the end). They can be in one class or different classes, use comments to make them clear.